### DISCUSSION TITLE
Error appending an item (uuid) to a list stored in an array (uuid[])

### DISCUSSION THREAD

**Discussion by MikeBrock03** [supabase/supabase#13174] [Questions]
I am having an issue with editing an item of my table. The architecture of the table is a collection of "profiles", containing uuid from auth, text for email, like: 
<img width="747" alt="Screenshot 2023-03-23 at 3 11 03 PM" src="https://user-images.githubusercontent.com/36043589/227376226-54d78968-5db0-4340-b878-cca117a08d96.png">

I want to be able to have a "friends" column, where users can add other users uuid to refer back to, like: 
<img width="643" alt="Screenshot 2023-03-23 at 3 11 33 PM" src="https://user-images.githubusercontent.com/36043589/227376441-6aeb5fae-3194-477a-8e54-703a89ee82bc.png">

This is the code that should execute to append the friend to the active user, and all of the console.logs return successfully: 

` 
    

    async function appendToArray(id, newItem) {

    console.log("id: ", id);
    console.log("friend id: ", newItem);

    try {
      const { data, error } = await supabase
        .from("profiles")
        .select("friends")
        .eq("id", id)
        .limit(1);
      if (error) {
        throw error;
      }

      // Check if the fetched items array is null (empty), if so, initialize it as an empty array
      const currentItems = data;
      const updatedArray = [...currentItems, newItem];
      const { updateError } = await supabase
        .from("profiles")
        .update({ friends: updatedArray })
        .eq("id", id);
      if (updateError) {
        throw updateError;
      }
     console.log("Item appended successfully!");
    } catch (error) {
      console.error("Error appending item to array:", error.message);
    }
  }
`

Is there something wrong with trying to "update" an item array in a table? What should I be doing different to access my table? 

Thank you!

**Comment by GaryAustin1** [comment]
Do you have RLS enabled?  If so do you meet both select and update policies?

**Reply by GaryAustin1** [reply]
Your error handling for the update does not look correct.  You should use const {error:updateError } = await update(...)
You currently will never get an error.
You should also console.log(data) before you do the update to see what it looks like.  It should be an array of objects with {colname:'uuid'} and that is not what you want to upload into a UUID array.  You just want UUID strings.

Also please don't generate new answers when responding to a previous response.

**Comment by MikeBrock03** [comment]
Thanks for the quick reply! For testing purposes, I’ve had it turned off.

**Comment by MikeBrock03** [comment]
Bumping this issue as it is still not resolved!
